README.md

```diff
@@ -1,3 +1,191 @@
+allowed to perform some action using some resource. Kafka Brokers by default use an ACL based mechanism where access rules are saved in ZooKeeper 
+and replicated across brokers. 
+
+Authorization in Kafka is implemented completely separately and independently of authentication. Thus, it is possible to configure Kafka Brokers 
+to use OAuth2 based authentication, and at the same time the default ACL authorization. 
+
+Since OAuth2 access tokens can contain arbitrary claims - including sets of roles or permissions, rather than basing authorization decisions purely 
+on user's identity, it could be based on externally assigned roles. All we need is define a mapping between access token claims and actions performed 
+on resources. For example, a 'kafka-topic:clicks_*:consumer' claim can be interpreted to mean that the user can read from any topic of which name 
+starts with 'clicks_'.
+
+Another possibility is to delegate authorization decisions to external authorization service - this can be a service provided by the same 
+OAuth2 authorization server used for authentication or it can be a different one, sharing the users database. 
+See https://docs.kantarainitiative.org/uma/wg/oauth-uma-grant-2.0-09.html for OAuth2 based standard for standardised UMA authorization services.
+
+At this time Strimzi Kafka OAuth doesn't provide authorization.
+
+
+
+Kafka OAuth2 Support
+--------------------
+Kafka comes with basic OAuth2 support in the form of SASL based authentication module which provides client-server retrieval, exchange and validation 
+of access token as credentials. For real world usage, extensions have to be provided in the form of JAAS callback handlers which is what Strimzi Kafka 
+OAuth does.
+
+
+Configuring the Kafka Broker as a Server
+----------------------------------------
+- Enable OAUTHBEARER as SASL mechanism
+- Configure `io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler` as validator callback handler
+- Specify `admin.user` - in order to properly integrate with default ACL authorization implementation, the broker authenticates itself as _admin_ user.
+With OAuth2 every access token is associated with some user via unique user identifier. `admin.user` should be set to the principal kafka-broker 
+authenticates as. 
+
+Specify valid issuer uri (OAUTH_VALID_ISSUER_URI) - only access tokens issued by this issuer will be accepted.
+
+Next, if your authorization server issues JWT access tokens, and provides a JWKS certificates endpoint, you can configure that for fast local token 
+validation (OAUTH_JWKS_ENDPOINT_URI). You can control how often JWKS certificates are refreshed by setting OAUTH_JWKS_REFRESH_SECONDS, and you can set
+the maximum time JWKS certificates are considered valid (OAUTH_JWKS_EXPIRY_SECONDS).
+
+TODO: Alternatively, you can also manually retrieve the public key from your authorization server, and configure it explicitly (OAUTH_JWKS_CERT).
```

Comment:
Maybe let's just remove it for now. It seemed like something one might easily implement, but it may not be entirely trivial working out storage format for the key and loading / translation to same classes used with JWKS endpoint. At this point it's low priority - I imagine it would only be needed for authorization servers that don't provide JWKS endpoint but support signed access tokens. Might be that there are no such server out there?